content/post/2019-12-10-pkg-urls.Rmarkdown

```diff
@@ -0,0 +1,159 @@
+knitr::opts_chunk$set(echo = TRUE,
+                      cache = TRUE,
+                      warning = FALSE)
+```
+
+Among DESCRIPTION usual fields is the free-text `URL` field where package authors can store various links: to the development website, docs, upstream tool, etc. In this post, we shall explain why storing URLs in DESCRIPTION is important, where else you should add URLs and what kind of URLs are stored in CRAN packages these days.
+
+## Why put URLs in DESCRIPTION?
+
+In the following we'll assume your package has some sort of online development repository ([GitHub](https://happygitwithr.com/big-picture.html)? [GitLab](https://gitlab.com/HeidiSeibold/setup-git-rstudio-gitlab#setup-git-rstudio-gitlab)? R-Forge?) and a documentation website (handily created via [pkgdown](https://pkgdown.r-lib.org/)?). Adding URLs to your package's online homes is extremely useful for several reasons.
+
+> As a side note: Yes, you can store several URLs under URL, even if the field name is singular. [See for instance  `rhub`'s DESCRIPTION](https://github.com/r-hub/rhub/blob/c51e0704ae7011536757f151144415323f4d77b9/DESCRIPTION#L15) :link: :link:
+
+```
+URL: https://github.com/r-hub/rhub, https://r-hub.github.io/rhub/
+```
+
+Why put URLs in DESCRIPTION?
+
+* It will help your users find your package's pretty documentation from the CRAN page, instead of just the less pretty PDF manual.
+
+* Likewise, from the CRAN page your contributors can directly find where to submit patches. 
+
+* If your package has a package-level man page, and it should (e.g. as drafted by [`usethis::use_package_doc()`](https://usethis.r-lib.org/reference/use_package_doc.html) and then generated by [`roxygen2`](https://roxygen2.r-lib.org/articles/rd.html#packages)), then after typing say `library("rhub")` and then `?rhub`, your users will find the useful links.
+
+* Other tool such as [`helpdesk`](https://github.com/yonicd/helpdesk) and [`pkgsearch` RStudio addin](https://r-hub.github.io/pkgsearch/reference/index.html#section-rstudio-addin) can help surface the URLs you store in DESCRIPTION.
+
+* Indirectly, having a link to the docs website and development repo will increase their page rank, [see useful comments in this discussion](https://community.rstudio.com/t/pkgdown-site-seo/26706), so potential users and contributors find them more easily by simply searching for your package.
+
+> Quick tip, you can add GitHub URLs (URL and BugReports) to DESCRIPTION by running [`usethis::use_github_links()`](https://usethis.r-lib.org/reference/use_github_links.html). :rocket:
+
+## Where else put your URLs?
+
+For the same reasons as previously, you should make the most of all places that can store your package's URL(s). Have you put your package's docs URL
+
+* in [the pkgdown config file](https://pkgdown.r-lib.org/reference/build_site.html#yaml-config), if that's how you built it?
+
+* in the GitHub repo URL field, or the equivalent for your development platform?
```

Comment:
I didn't immediately know what this was, but maybe that's just me.

Reply:
good point, especially as it's not obvious where that is in the docs!

Reply:
This is the URL at the top of the repo page, right? Maybe we can have a screenshot here, or just write "at the top right of the repo page" or sth similar?